src/prog_models/prognostics_model.py

```diff
@@ -1201,58 +1204,96 @@ def estimate_params(self, runs : List[tuple] = None, keys : List[str] = None, ti
         """
-        from scipy.optimize import minimize
+        from scipy.optimize import minimize, show_options
 
         if keys is None:
             # if no keys provided, use all
             keys = [key for key in self.parameters.keys() if isinstance(self.parameters[key], Number)]
+        
+        for key in keys:
+            if key not in self.parameters:
+                raise ValueError("Passing in key that does not exist.")
 
         config = {
             'method': 'nelder-mead',
             'bounds': tuple((-np.inf, np.inf) for _ in keys),
-            'options': {'xatol': 1e-8},
+            'options': {'xatol': 1e-8}
         }
         config.update(kwargs)
 
+        if isinstance(times, set) or isinstance(inputs, set) or isinstance(outputs, set):
+            raise TypeError(f"Times, Inputs, and Outputs cannot be a Set. Sets are unordered by definition, so passing in arguments as Sets may have undefined behavior.")
+
+        # if parameters not in parent wrapper sequence, then place them into one.
+        if not runs and times and inputs and outputs:
+            if not isinstance(times[0], (Sequence, np.ndarray)):
+                times = [times]
+            if not isinstance(inputs[0], (Sequence, np.ndarray)):
+                inputs = [inputs]
+            if not isinstance(outputs[0], (Sequence, np.ndarray)):
+                outputs = [outputs]
+
         if runs is None and (times is None or inputs is None or outputs is None):
-            raise ValueError("Must provide either runs or times, inputs, and outputs")
+            if times is None and inputs is None and outputs is None:
+                # estimate paramters requires times, inputs and outputs. Missing x
+                raise ValueError("Must pass in times, inputs, and outputs")
+            if times is None and inputs is None:
+                raise ValueError("Must pass in times and inputs")
+            if times is None and outputs is None:
+                raise ValueError("Must pass in times and outputs")
+            if inputs is None and outputs is None:
+                raise ValueError("Must pass in inpupts and outputs")
+            if times is None:
+                raise ValueError("Must pass in times")
+            if inputs is None:
+                raise ValueError("Must pass in inputs")
+            if outputs is None:
+                raise ValueError("Must pass in outputs")
+            raise ValueError("Must provide either runs")
         if runs is None:
             if len(times) != len(inputs) or len(outputs) != len(inputs):
-                raise ValueError("Times, inputs, and outputs must be same length")
+                raise ValueError(f"Times, inputs, and outputs must be same length. Length of times: {len(times)}, Length of inputs: {len(inputs)}, Length of outputs: {len(outputs)}")
+            if len(times) == 0:
+                # since they are all the same length, does not matter as much
```

Comment:
Can we change this comment to be more descriptive of what we are checking for here? (Or delete it, I don't think it's necessary)

Reply:
What about:

```python
# Since inputs, times, and outputs are already confirmed to be the same length, only check that one is not empty
```